main/inc/lib/hook/HookEvent.class.php

```diff
@@ -0,0 +1,209 @@
+<?php
+/* For licensing terms, see /license.txt */
+
+
+abstract class HookEvent implements HookEventInterface
+{
+    public $observers;
+    public $eventName;
+    public $eventData;
+
+    /**
+     * Construct Method
+     * @param $eventName
+     * @throws Exception
+     */
+    protected function __construct($eventName)
+    {
+        if (self::isHookPluginActive()) {
+            $this->observers = new SplObjectStorage();
+            $this->eventName = $eventName;
+            $this->eventData = array();
+            $this->plugin = HookManagement::create();
+            $this->loadAttachments();
+        } else {
+            throw new \Exception('Hook Management Plugin is not active');
+        }
+    }
+
+    /**
+     * Return the singleton instance of Hook event.
+     * If Hook Management plugin is not enabled, will return NULL
+     * @return HookEventInterface|null
+     */
+    public static function create()
+    {
+        static $result = null;
+
+        if ($result) {
+            return $result;
+        } else {
+            try {
+                $class = get_called_class();
+                return new $class;
+            } catch (Exception $e) {
+                return null;
+            }
+        }
+    }
+
+    /**
+     * Attach an HookObserver
+     * @link http://php.net/manual/en/splsubject.attach.php
+     * @param \HookObserverInterface| $observer <p>
+     * The <b>HookObserver</b> to attach.
+     * </p>
+     * @return void
+     */
+    public function attach(HookObserverInterface $observer)
+    {
+        global $_hook;
```

Comment:
The name should be $hook.
You can setup the variable with 
setHook($hook)  and the call it with getHook()